### PR TITLE
 chore(vendor): Migrating to go modules from vendor

### PR DESCRIPTION
**Pull request by mynktl**
Changes:
- Migrating to go modules from vendor
- Updating go version to 1.14.x in travis build.

Commit to review : https://github.com/openebs/velero-plugin/pull/68/commits/65597f5f65cd6e090a729c162ccebdf088840486

Signed-off-by: mayank <mayank.patel@mayadata.io>

### CODE REVIEW

**Comment by kmova** [comment]
@mynktl  -- can you update the PR description with specific commits that should be reviewed. 

Also if possible push the files that need to be reviewed into different commit than the vendor code update commit.

**Comment by mynktl** [comment]
Hi @kmova 
I've updated the description with commit details.
Just to summarize the commits , There are three commits : 
https://github.com/openebs/velero-plugin/pull/68/commits/92cf52bbf04839bf956369bbcd80374b8c152430 : Migrating to go modules from vendor : which is about updating to the module
https://github.com/openebs/velero-plugin/pull/68/commits/65597f5f65cd6e090a729c162ccebdf088840486 :  fixing linting issues , which have changes about fixes for linting issue
https://github.com/openebs/velero-plugin/pull/68/commits/b938088f4cb2f561cc487b331416d98d3e13efb7:  fixing travis file , it fixes the travis file to trigger the travis build.

File: Makefile [hunk 1]
```diff
@@ -41,72 +35,29 @@ endif
...
-	cp Dockerfile _output/Dockerfile
+	@cp Dockerfile _output/Dockerfile
 	docker build -t $(IMAGE):$(IMAGE_TAG) --build-arg BUILD_DATE=${BUILD_DATE} -f _output/Dockerfile _output
 
-all-ci: $(addprefix ci-, $(BIN))
-
-ci-%:
-	$(MAKE) --no-print-directory BIN=$* ci
-
 ci:
```
Comment (prateekpandey14) [inline Makefile:44]: i think target name should be `build` instead of `ci` ?

File: Makefile [hunk 2]
```diff
@@ -41,72 +35,29 @@ endif
...
-	-gometalinter.v1 --config=metalinter.config ./...
-
-golint:
-	@gometalinter.v1 --install
-	@gometalinter.v1 --vendor --disable-all -E errcheck -E misspell ./...
+# Run linter using docker image
+lint-docker:
+	@docker run -i								\
+		--rm -v $$(pwd):/app -w /app 			\
+		golangci/golangci-lint:v1.24.0 			\
```
Comment (prateekpandey14) [inline Makefile:52]: nit : makefile indentation `\`

File: Makefile [hunk 3]
```diff
@@ -41,72 +35,29 @@ endif
...
+		bash -c "GOGC=75 golangci-lint run"
 
-check: golint-travis format vet
+# Run linter using local binary
+lint:
+	@golangci-lint run
```
Comment (prateekpandey14) [inline Makefile:58]: We can also add the `deps` target to manage the vendor dependencies easily
```
.PHONY: deps
deps: ## Ensures fresh go.mod and go.sum.
	@go mod tidy
	@go mod verify
```

Comment (mynktl) [reply Makefile:58]: when i ran go mod tidy, i got error
```
github.com/openebs/velero-plugin/pkg/clouduploader imports
	gocloud.dev/blob/gcsblob tested by
	gocloud.dev/blob/gcsblob.test imports
	gocloud.dev/internal/testing/setup imports
	github.com/Azure/azure-storage-blob-go/azblob tested by
	github.com/Azure/azure-storage-blob-go/azblob.test imports
	github.com/Azure/go-autorest/autorest/adal: ambiguous import: found package github.com/Azure/go-autorest/autorest/adal in multiple modules:
	github.com/Azure/go-autorest v12.0.0+incompatible (/home/mayank/go/pkg/mod/github.com/!azure/go-autorest@v12.0.0+incompatible/autorest/adal)
	github.com/Azure/go-autorest/autorest/adal v0.5.0 (/home/mayank/go/pkg/mod/github.com/!azure/go-autorest/autorest/adal@v0.5.0)
github.com/openebs/velero-plugin/velero-blockstore-cstor imports
	github.com/vmware-tanzu/velero/pkg/plugin/framework tested by
	github.com/vmware-tanzu/velero/pkg/plugin/framework.test imports
	github.com/vmware-tanzu/velero/pkg/test imports
	github.com/vmware-tanzu/velero/pkg/client imports
	k8s.io/client-go/plugin/pkg/client/auth/azure imports
	github.com/Azure/go-autorest/autorest: ambiguous import: found package github.com/Azure/go-autorest/autorest in multiple modules:
	github.com/Azure/go-autorest v12.0.0+incompatible (/home/mayank/go/pkg/mod/github.com/!azure/go-autorest@v12.0.0+incompatible/autorest)
	github.com/Azure/go-autorest/autorest v0.9.0 (/home/mayank/go/pkg/mod/github.com/!azure/go-autorest/autorest@v0.9.0)
github.com/openebs/velero-plugin/velero-blockstore-cstor imports
	github.com/vmware-tanzu/velero/pkg/plugin/framework tested by
	github.com/vmware-tanzu/velero/pkg/plugin/framework.test imports
	github.com/vmware-tanzu/velero/pkg/test imports
	github.com/vmware-tanzu/velero/pkg/client imports
	k8s.io/client-go/plugin/pkg/client/auth/azure imports
	github.com/Azure/go-autorest/autorest/azure: ambiguous import: found package github.com/Azure/go-autorest/autorest/azure in multiple modules:
	github.com/Azure/go-autorest v12.0.0+incompatible (/home/mayank/go/pkg/mod/github.com/!azure/go-autorest@v12.0.0+incompatible/autorest/azure)
	github.com/Azure/go-autorest/autorest v0.9.0 (/home/mayank/go/pkg/mod/github.com/!azure/go-autorest/autorest@v0.9.0/azure)
```
but go mod verify return 0.
```
all modules verified
```
What could be the reason?

Comment (mynktl) [reply Makefile:62]: it happened due to conflict. Importing ambiguous import package with `go get package@none`  fix the issue.

**Comment by kmova** [comment]
@mynktl -- can you rebase the commits in two commits so that we can rebase and merge this PR. 

- vendor files commit
- rest of the changes to migrate to go mod. 

If we do squash merge into one commit, the commit will have too many files.

**Comment by mynktl** [comment]
@kmova I've forced push with 2 commits. Now we are good to merge.

**Pull request merged**